### PR TITLE
DO NOT MERGE:  m10x day 1 updates to transport interface

### PR DESCRIPTION
**Pull request by TimEvens**
Do not merge yet. This is for review right now.

### CODE REVIEW

File: include/transport/transport.h [hunk 1]
```diff
@@ -4,90 +4,171 @@
...
+
+/**
+ * Transport configuration parameters
+ */
+struct TransportConfig {
+	// Nothing yet
+};
+
+/**
+ * @brief IP Transport interface
```
Comment (suhasHere) [inline include/transport/transport.h:65]: nit s/IP/ITranspor

File: include/transport/transport.h [hunk 2]
```diff
@@ -4,90 +4,171 @@
...
 class ITransport {
 public:
-  /*
+  /**
    * @brief Async Callback API on the transport
    */
-  struct TransportDelegate {
+  class TransportDelegate {
+	public:
     virtual ~TransportDelegate() = default;
```
Comment (suhasHere) [inline include/transport/transport.h:89]: nit .. please run ```make format``` before merge .. thanks

File: include/transport/transport.h [hunk 3]
```diff
@@ -4,90 +4,171 @@
...
+    /**
      * @brief Report arrival of a new connection
-     * @param context_id : Transport context identifier mapped to the connection
+     *
+     * @param[in] context_id	Transport context identifier mapped to the connection
+     * @param[in] remote			Transport information for the connection
      */
-    virtual void on_new_connection(const TransportContextId &context_id) = 0;
+    virtual void on_new_connection(const TransportContextId &context_id,
+																	 const TransportRemote remote) = 0;
```
Comment (suhasHere) [inline include/transport/transport.h:110]: s/TransportRemote/TransportRemote&

**Review by suhasHere — APPROVED** [review]
few nits and formatting fixes. Otherwise LGTM

File: include/transport/transport.h [hunk 4]
```diff
@@ -3,118 +3,246 @@
...
+  CannotResolveHostname,
+  InvalidIpv4Address,
+  InvalidIpv6Address
+};
+
+/**
+ *
+ */
+enum class TransportProtocol { UDP = 0, QUIC };
+
```
Comment (suhasHere) [inline include/transport/transport.h:43]: thinking about it let's make it
`TransportProtocol { UDP=0, PICOQUIC} ` to make it easy to plugin in

File: include/transport/transport.h [hunk 5]
```diff
@@ -3,118 +3,246 @@
...
+   * @details Data received by the transport will be queued and made available
+   * to the caller using this method.  An empty return will be
+   *
+   * @param[in] context_id		Identifying the connection
+   * @param[in] mStreamId			Media stream Id to receive data
+   * from
    */
   virtual std::optional<std::vector<uint8_t>>
-  dequeue(const TransportContextId &tcid, const MediaStreamId &msid) = 0;
+  dequeue(const TransportContextId &context_id,
```
Comment (fluffy) [inline include/transport/transport.h:244]: should context_id change to conectId everywhere ?

File: include/transport/transport.h [hunk 6]
```diff
@@ -3,118 +3,246 @@
 #include <mutex>
 #include <optional>
 #include <queue>
+#include <sys/socket.h>
```
Comment (fluffy) [inline include/transport/transport.h:6]: This seems like we have the wrong level of abstraction if we need socket here

Comment (fluffy) [reply include/transport/transport.h:6]: actually, I can't see where we need this in this header file

File: include/transport/transport.h [hunk 7]
```diff
@@ -3,118 +3,246 @@
...
+    uint64_t; ///< Context Id is a 64bit number that is used as a key to maps
+using MediaStreamId = uint64_t; ///< Media stream Id is a 64bit number that is
+                                ///< used as a key to maps
 
-enum class TransportStatus : uint8_t { Ready = 0, Disconnected };
+/**
+ * Transport status/state values
+ */
+enum class TransportStatus : uint8_t {
+  Ready = 0,
```
[the remark this answers is not in the document]
Comment (fluffy) [reply include/transport/transport.h:20]: on 2nd thought, given this is a status for servers that are listening too, this seems like a bad idea.

File: include/transport/transport.h [hunk 8]
```diff
@@ -3,118 +3,246 @@
...
+  Connecting,
+  RemoteRequestClose,
+  Disconnected
+};
```
Comment (fluffy) [inline include/transport/transport.h:25]: I'm not sure these states make sense to describe a server that at the same time lilstening for new connection, fully connected to some client, and in the process of connecting to another, and some other client has already disconnected

File: include/transport/transport.h [hunk 9]
```diff
@@ -3,118 +3,246 @@
...
      *
-     * param context_id : Transport context identifier mapped to the connection
+     * @details Applications must invoke ITransport::deqeue() to obtain
+     * 		the data by passing the transport context id
+     *
+     * @param[in] context_id 	Transport context identifier mapped to the
+     * connection
      */
-    virtual void on_recv_notify(TransportContextId &tcid) = 0;
+    virtual void on_recv_notify(const TransportContextId &context_id) = 0;
```
Comment (fluffy) [inline include/transport/transport.h:137]: I'm a bit lost on mediaID but wondering if this also needs an mediaID

**Review by fluffy — CHANGES_REQUESTED** [review]
Few minor things

**Comment by TimEvens** [comment]
Review done, closing and will open a new PR that can be merged.

**Pull request closed without merging**